manifests/postgres.pp

```diff
@@ -173,7 +173,7 @@
   # define user used by Barman to connect into PostgreSQL database(s)
   postgresql::server::role { $barman_dbuser:
     login         => true,
-    password_hash => postgresql_password($barman_dbuser, $real_password),
+    password_hash => postgresql_password($barman_dbuser, inline_template('<%= @real_password.to_s %>')),
```

Comment:
The 'problem' just has to do with changes that were made in puppet >=4.0.  Not actually trying to make anything different happen, just making the syntax work with newer version of puppet and puppet-lint.